src/deno.rs

```diff
@@ -0,0 +1,375 @@
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(f, "workspace:{}", self.0)
+  }
+}
+
+impl Serialize for WorkspacePackageNvReference {
+  fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+  where
+    S: serde::Serializer,
+  {
+    serializer.serialize_str(&self.to_string())
+  }
+}
+
+impl<'de> Deserialize<'de> for WorkspacePackageNvReference {
+  fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+  where
+    D: serde::Deserializer<'de>,
+  {
+    let text = String::deserialize(deserializer)?;
+    match Self::from_str(&text) {
+      Ok(req) => Ok(req),
+      Err(err) => Err(serde::de::Error::custom(err)),
+    }
+  }
+}
+
+#[cfg(test)]
+mod test {
+  use super::*;
+
+  #[test]
+  fn deno_req_ref() {
+    {
+      let req_ref = DenoPackageReqReference::from_specifier(
+        &Url::parse("deno:/foo").unwrap(),
+      )
+      .unwrap();
+      assert_eq!(req_ref.req().name, "foo");
+      assert_eq!(req_ref.req().version_req.to_string(), "*");
+      assert_eq!(req_ref.sub_path(), None);
+    }
+    {
+      let req_ref =
+        DenoPackageReqReference::from_str("deno:foo@1/mod.ts").unwrap();
+      assert_eq!(req_ref.req().name, "foo");
+      assert_eq!(req_ref.req().version_req.to_string(), "1");
+      assert_eq!(req_ref.sub_path(), Some("mod.ts"));
+    }
+    {
+      let req_ref =
+        DenoPackageReqReference::from_str("deno:@scope/foo@^1.0.0/mod.ts")
+          .unwrap();
+      assert_eq!(req_ref.req().name, "@scope/foo");
+      assert_eq!(req_ref.req().version_req.to_string(), "^1.0.0");
+      assert_eq!(req_ref.sub_path(), Some("mod.ts"));
+    }
+  }
+
+  #[test]
+  fn deno_nv_ref() {
+    {
+      let nv_ref = DenoPackageNvReference::from_specifier(
+        &Url::parse("deno:/foo@1.0.0").unwrap(),
+      )
+      .unwrap();
+      assert_eq!(nv_ref.nv().name, "foo");
+      assert_eq!(nv_ref.nv().version.to_string(), "1.0.0");
+      assert_eq!(nv_ref.sub_path(), None);
+      assert_eq!(nv_ref.as_specifier().as_str(), "deno:/foo@1.0.0");
+    }
+    {
+      let nv_ref =
+        DenoPackageNvReference::from_str("deno:foo@1.0.0/mod.ts").unwrap();
+      assert_eq!(nv_ref.nv().name, "foo");
+      assert_eq!(nv_ref.nv().version.to_string(), "1.0.0");
+      assert_eq!(nv_ref.sub_path(), Some("mod.ts"));
+      assert_eq!(nv_ref.as_specifier().as_str(), "deno:/foo@1.0.0/mod.ts");
+    }
+    {
+      let nv_ref =
+        DenoPackageNvReference::from_str("deno:@scope/foo@1.0.0/mod.ts")
+          .unwrap();
+      assert_eq!(nv_ref.nv().name, "@scope/foo");
+      assert_eq!(nv_ref.nv().version.to_string(), "1.0.0");
+      assert_eq!(nv_ref.sub_path(), Some("mod.ts"));
+      assert_eq!(
+        nv_ref.as_specifier().as_str(),
+        "deno:/@scope/foo@1.0.0/mod.ts"
+      );
+    }
+  }
+
+  #[test]
+  fn workspace_req_ref() {
+    {
+      let req_ref = WorkspacePackageReqReference::from_specifier(
+        &Url::parse("workspace:/foo").unwrap(),
```

Comment:
So we're gonna require `:/` in contrast to `npm:` specifiers?

Reply:
I was just testing this out. Right now it doesn't require the slash.